### PR TITLE
fix : issues #3121 & #3354 - adding options for dropdown positionning…

### PR DESCRIPTION
**Pull request by DVD27**
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made :

- Add an option for choice the dropdown position on open (above, below or both) (issues #3121)
- Add an option for choice what to do when backspace event is produce on multiple select (unselect (delete just the tag), none (do nothing), text (transform the tag to text like today)) (issues #3354)

### CODE REVIEW

File: dist/js/select2.full.js [hunk 1]
```diff
@@ -30,13 +30,11 @@
...
 if (!S2) { S2 = {}; } else { require = S2; }
 /**
- * @license almond 0.3.1 Copyright (c) 2011-2014, The Dojo Foundation All Rights Reserved.
- * Available via the MIT or new BSD license.
- * see: http://github.com/jrburke/almond for details
+ * @license almond 0.3.2 Copyright jQuery Foundation and other contributors.
+ * Released under MIT license, http://github.com/requirejs/almond/LICENSE
  */
 //Going sloppy to avoid 'use strict' string cost, but strict practices should
 //be followed.
```
Comment (LayZeeDK) [inline dist/js/select2.full.js]: If you are removing the `sloppy` JSLint option, you should remove this comment as well.

Comment (kevin-brown) [inline dist/js/select2.full.js]: This is a part of the built JS files, and this PR upgraded Almond in the process (which is where this comment comes from).

**Comment by Ivanca** [comment]
It seems you are writing `dropdownLocation` in some parts where it should be `dropdownPosition`

**Comment by DVD27** [comment]
Thanks Ivanca (I don't have see it !), I have correct this issues in the new pull-request.

**Pull request closed without merging**